apps/wallet-dashboard/hooks/useGetStardustIndexerOutputs.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { useQuery } from '@tanstack/react-query';
+import { useGetCurrentEpochStartTimestamp } from '@/hooks';
+import {
+    mapStardustBasicOutputs,
+    mapStardustNftOutputs,
+    StardustIndexerOutput,
+    TimeUnit,
+    useStardustIndexerClientContext,
+} from '@iota/core';
+import { IotaObjectData } from '@iota/iota-sdk/client';
+
+const LIMIT_PER_REQ = 50;
+
+export function useGetStardustIndexerOutputs(address: string) {
+    const { data: currentEpochMs } = useGetCurrentEpochStartTimestamp();
+    const { stardustIndexerClient } = useStardustIndexerClientContext();
+
+    const fetchPaginatedOutputs = async (
+        mapFn: (output: StardustIndexerOutput) => IotaObjectData,
+        fetchFn?: (
+            address: string,
+            params: { page: number; limit: number },
+        ) => Promise<StardustIndexerOutput[]>,
+    ) => {
+        const allOutputs = [];
+        let page = 1;
+        let hasMoreData = true;
+
+        if (!fetchFn) {
+            return [];
+        }
+
+        try {
+            while (hasMoreData) {
+                const outputs = await fetchFn(address, { page, limit: LIMIT_PER_REQ });
+
+                if (!outputs || outputs.length === 0) {
+                    hasMoreData = false;
+                } else {
+                    allOutputs.push(...outputs);
+                    page++;
+                }
+            }
+        } catch (e) {
+            console.error(e);
+        }
+
+        return allOutputs.map(mapFn);
+    };
+
+    return useQuery({
+        queryKey: [
+            'stardust-indexer-outputs',
+            address,
+            currentEpochMs,
+            stardustIndexerClient?.getBasicResolvedOutputs,
+            stardustIndexerClient?.getNftResolvedOutputs,
+        ],
+        queryFn: async () => {
+            const basicObjects = await fetchPaginatedOutputs(
+                mapStardustBasicOutputs,
+                stardustIndexerClient?.getBasicResolvedOutputs,
+            );
+
+            const nftObjects = await fetchPaginatedOutputs(
+                mapStardustNftOutputs,
+                stardustIndexerClient?.getNftResolvedOutputs,
+            );
```

Comment:
Same as the previous comment about basic outputs, but for nfts

Reply:
Misunderstanding on my side, getting the information with `resolved` is the correct thing to do